rskj-core/fuzzing.gradle

```diff
@@ -0,0 +1,583 @@
+sourceSets {
+    fuzzTest {
+        java {
+            srcDirs = ['src/fuzzTest/java']
+        }
+        resources {
+            srcDirs = ['src/fuzzTest/resources']
+        }
+        compileClasspath += sourceSets.main.output + sourceSets.test.output
+        runtimeClasspath += sourceSets.main.output + sourceSets.test.output
+    }
+}
+
+configurations {
+    fuzzTestImplementation.extendsFrom testImplementation
+    fuzzTestRuntimeOnly.extendsFrom testRuntimeOnly
+}
+
+task fuzzTest(type: Test) {
+    useJUnitPlatform()
+    testClassesDirs = sourceSets.fuzzTest.output.classesDirs
+    classpath = sourceSets.fuzzTest.runtimeClasspath
+
+    beforeTest { descriptor ->
+        logger.lifecycle("Running fuzz test: ${descriptor}")
+    }
+
+    jvmArgs '-Xss32m', '-Xmx3G'
+
+    testLogging {
+        events "failed"
+        exceptionFormat "short"
+    }
+
+    outputs.upToDateWhen { false }
+
+    doLast {
+        println "\n<################# FUZZING REPORT ####################>"
+        println "Fuzzing report generated at: ${reports.html.destination}/index.html"
+        println "<##############################################>\n"
+    }
+}
+
+task fuzzCoverageTestReport(type: JacocoReport) {
+    dependsOn fuzzTest
+    executionData(fileTree(buildDir).include("**/jacoco/*.exec"))
+
+    sourceSets sourceSets.main, sourceSets.fuzzTest
+
+    reports {
+        xml.required = true
+        html.required = true
+    }
+
+    doLast {
+        println "\n<################# COVERAGE REPORT ###################>"
```

Comment:
I would update this messages to say something like "FUZZ COVERAGE REPORT" maybe it's not needed, but technically that's the coverage report we're generating here.